.github/workflows/docker.yml

```diff
@@ -60,6 +60,19 @@ jobs:
           TAG_LATEST=${{(github.event_name == 'push' && github.ref == 'refs/heads/main') && 'true' || 'false'}}
           echo "TAG_LATEST=${TAG_LATEST}" >> $GITHUB_ENV
 
+          # Cache params are a bit of a pain
+          cachefor () {
+              echo "$1.cache-from=type=local,src=/tmp/.buildx-cache/$1"
+              echo "$1.cache-to=type=local,dest=/tmp/.buildx-cache-new/$1"
+          }
+
+          echo 'cache_params<<EOF' >> $GITHUB_OUTPUT
+          for img in server admin-tools auto-setup; do
+            cachefor $img >> $GITHUB_OUTPUT
+          done
+          echo 'EOF' >> $GITHUB_OUTPUT
+          cat $GITHUB_OUTPUT
```

Comment:
Fun fact: this grossness is how you get a multiline output parameter

Reply:
What it's doing is generating the following code (which is how you do this in a shell):
```
cache_params<<EOF
BODY CONTENT
EOF
```

Reply:
I really don't want to know and understand this!